### PR TITLE
AdminController

### PR DESCRIPTION
**Pull request by Zarathustra4**

### CODE REVIEW

File: src/main/java/com/passpnu/passwordmanager/controller/AdminController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            return new ResponseEntity<>("Username is already taken", HttpStatus.BAD_REQUEST);
+        }
+
+        userEntityService.saveAdmin(userDto);
+
+        return new ResponseEntity<>("User registered successfully", HttpStatus.OK);
+    }
+
+    @PutMapping("/change-role")
+    public ResponseEntity<String> changeRole(@RequestBody ChangeRoleDto changeRoleDto) throws NameNotFoundException {
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/AdminController.java]: Create own exception instead

File: src/main/java/com/passpnu/passwordmanager/controller/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.naming.NameNotFoundException;
+
+@RestController
+@RequestMapping("/admin")
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/AdminController.java:19]: If this controller is just for admin use, it should be annotated with `@PreAuthorize` or configured via security config

File: src/main/java/com/passpnu/passwordmanager/controller/AdminController.java (hunk 1 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/AdminController.java]: fixed

File: src/main/java/com/passpnu/passwordmanager/controller/AdminController.java (hunk 2 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/AdminController.java:19]: fixed

**Review by justme1135 — APPROVED** [review]

**Pull request merged**